src/local_scheduler/local_scheduler.cc

```diff
@@ -617,12 +631,17 @@ void finish_task(LocalSchedulerState *state,
       int task_state =
           actor_checkpoint_failed ? TASK_STATUS_LOST : TASK_STATUS_DONE;
       Task_set_state(worker->task_in_progress, task_state);
+#if !RAY_USE_NEW_GCS
       task_table_update(state->db, worker->task_in_progress, NULL, NULL, NULL);
-      /* The call to task_table_update takes ownership of the
-       * task_in_progress, so we set the pointer to NULL so it is not used. */
+#else
+      RAY_CHECK_OK(TaskTableAdd(&state->gcs_client, worker->task_in_progress));
+      Task_free(worker->task_in_progress);
+#endif
     } else {
       Task_free(worker->task_in_progress);
     }
+    /* The call to task_table_update takes ownership of the
```

Comment:
Not specifically about this PR, but for new code I would suggest switching to `//`.

Reply:
For the new C++ files I'm doing that, for existing ones I'm taking the style that is already there. I'll create a PR that switches them all at once.